### PR TITLE
Handle detecting docker bridge interfaces

### PR DESCRIPTION
**Pull request by bonidjukic**
It seems that if there are any docker network bridge interfaces
present on Linux (Unix?) machine (tested on Ubuntu 17.10), it makes
ifcfg crash while detecting devices due to the different naming
pattern.

As described here: https://github.com/docker/labs/blob/master/networking/concepts/05-bridge-networks.md
docker creates network bridge interfaces with the following pattern,
e.g.: `br-4bcc22f5e5b9`, which fails to match on any currently given
regexp pattern.

This commit simply adds hypen to the `UnixParser`, `get_patterns`'s
first regexp pattern in order to catch interfaces with hypens as well.

Also adds a test for this particular case.

### CODE REVIEW

File: src/ifcfg/parser.py [hunk 1]
```diff
@@ -209,7 +209,7 @@ def get_command(cls):
     @classmethod
     def get_patterns(cls):
         return [
-            '(?P<device>^[a-z-A-Z0-9:]+): flags=(?P<flags>.*) mtu (?P<mtu>.*)',
+            '(?P<device>^[-a-zA-Z0-9:]+): flags=(?P<flags>.*) mtu (?P<mtu>.*)',
```
Comment (benjaoming) [inline src/ifcfg/parser.py]: Hi @bonidjukic !

Is it that the dash in front has been added? Maybe safer to escape it?

Would be nice if you could provide the output that you're parsing in a test :)

Comment (bonidjukic) [reply src/ifcfg/parser.py]: Hi @benjaoming, thanks for the comments!

>Is it that the dash in front has been added? Maybe safer to escape it?

To be completely precise, it's hypen, but I guess that's what you meant. As mentioned before, hypen is added due to the docker bridge interfaces having them in their names.

As far as the escaping goes, my understanding is that the both cases are "safe" (having it as the first/last character in the group or escaping it with backlash.

Here's an SO question/answer which mentions this exact use case: https://stackoverflow.com/questions/9589074/regex-should-hyphens-be-escaped/9589642#9589642

>Would be nice if you could provide the output that you're parsing in a test :)
No sure what you meant here Ben, I added the output to the [tests/ifconfig_out.py](https://github.com/learningequality/python-ifcfg/pull/12/commits/a69bc51943409bbb43d42142369f7ab4719bb7c9#diff-b81024eb10fe13945e5e4801601e376e) or did you mean the parser output?

**Comment by benjaoming** [comment]
Oh just noticing one thing here... the PR is against our own fork.. you might as well point it upstream, that's where we collect, tag and release... not sure if we really need to maintain this repo anymore. Will check to see if we've pinned anything in requirements.txt on it...

**Comment by bonidjukic** [comment]
Yes, I wasn't sure if I should create it against our fork or directly to the upstream.

Cool, if we can work directly with the upstream, it would definitely make things easier - let me know and I'll move the PR there.

**Comment by jamalex** [comment]
Haha, nevermind, didn't realize @benjaoming is a core maintainer of the upstream repo. :) Yeah, would make sense to retarget there, and then shut down this fork.

**Comment by benjaoming** [comment]
Deleting the repo! KA Lite is pinned on PyPi releases since 0.16.

**Pull request closed without merging**